pkgs/tools/audio/beets/common.nix

```
@@ -63,6 +63,10 @@ python3Packages.buildPythonApplication rec {
     unidecode
   ] ++ (concatMap (p: p.propagatedBuildInputs) (attrValues enabledPlugins));
 
+  propagatedNativeBuildInputs = [
```

Did you also try `nativeBuildInputs`? Did you try also this and/or `propagatedNativeBuildInputs` with and without `gobject-introspection` in `propagatedBuildInputs`?

---

- `nativeBuildInputs` works as well
- without adding `nativeBuildInputs` or `propagatedNativeBuildInputs` gives the error in the issue linked
- having `nativeBuildInputs`/`propagatedNativeBuildInputs` and removing the gobject from `propagatedBuildInputs` gives the error in the issue linked

---

As I understand it (which is admittedly a fairly limited understanding), `gobject-introspection` needs to be in both `buildInputs` and `nativeBuildInputs` for stuff with `strictDeps = true`, like Python packages (see https://github.com/NixOS/nixpkgs/issues/56943#issuecomment-1131643663 and probably a lot of other mentions in other packages)

If it's supposed to be propagated for `buildInputs` (which seems to be common for Python libraries, but may be an antipattern) then it should probably be propagated for `nativeBuildInputs` too. But I don't think beets is ever used as a library so I'm uncertain whether it actually needs to be propagated by either

I do wonder if it should be propagated by `pygobject` itself instead of by every package that depends on `pygobject`, but that's way above my level and possibly a very bad idea. Things breaking now seems to have started with the last staging-next iteration, but I didn't find an immediately obvious change that started these failures

(If someone more intelligent than me has better info on `pyobject` and `gobject-introspection`, I would definitely appreciate information and/or clarification)

---

> But I don't think beets is ever used as a library so I'm uncertain whether it actually needs to be propagated by either

IIRC that beets is used as a dependency for building it's plugins, but few if any plugins require `gstreamer` related features. I think we should use `nativeBuildInputs` with a comment:

```suggestion
  # see: https://github.com/NixOS/nixpkgs/issues/56943#issuecomment-1131643663
  nativeBuildInputs = [
```

---

Do you think `gobject-introspection` should be removed from `propagatedBuildInputs` and just put in to `buildInputs` as well then?

---

> I think we should use `nativeBuildInputs` with a comment:

Did that in an amended commit.

---

> Do you think `gobject-introspection` should be removed from `propagatedBuildInputs` and just put in to `buildInputs` as well then?

Yes please! Good idea, that would be more accurate!